### ISSUE TITLE
ImportError: No module named '_pywrap_tensorflow_internal'

### ISSUE THREAD

**Issue by amtagrwl** [tensorflow/tensorflow#10111]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows 7 
- **TensorFlow installed from (source or binary)**:
Using pip3 
- **TensorFlow version (use command below)**:
1.1
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
8.0 
- **GPU model and memory**:
Quadro K620
- **Exact command to reproduce**:
import tensorflow as tf

I have installed everything following the instructions and opened a [ post on SO ](https://stackoverflow.com/questions/44080677/no-module-named-pywrap-tensorflow-internal) but haven't receieved many responses. I have looked at multiple other posts with the same keywords but was unable to solve the issue from there. I used the standard installation instructions using pip3 for windows. 

The following is the stacktrace when I run "import tensorflow as tf" in python command line. 

```
Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\aagarwal>python
Python 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AM
D64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
Traceback (most recent call last):
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow_internal.py", line 18, in swig_import_hel
per
    return importlib.import_module(mname)
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\importlib\_
_init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 914, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow.py", line 41, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow_internal.py", line 21, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow_internal.py", line 20, in swig_import_hel
per
    return importlib.import_module('_pywrap_tensorflow_internal')
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\importlib\_
_init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow_internal'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\__init__.py", line 51, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow.py", line 52, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow_internal.py", line 18, in swig_import_hel
per
    return importlib.import_module(mname)
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\importlib\_
_init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 914, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow.py", line 41, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow_internal.py", line 21, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\site-packag
es\tensorflow\python\pywrap_tensorflow_internal.py", line 20, in swig_import_hel
per
    return importlib.import_module('_pywrap_tensorflow_internal')
  File "C:\Users\aagarwal\AppData\Local\Programs\Python\Python35\lib\importlib\_
_init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow_internal'


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/install_sources#common_installation_probl
ems

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.
>>>
```

**Comment by gunan** [comment]
It is possible some of the files did not get installed properly.
Could you try reinstalling TF, maybe that will resolve the problem?

**Comment by eXcellme** [comment]
Same problem . 
Windows 10 / Python 3.5.3 / CPU mode

**Comment by amtagrwl** [comment]
Solved. 

Issue was with CuDNN version. Version 6.0 does not work. You need to install version 5.1.

**Comment by Idobenja** [comment]
@amtagrwl thanks, worked for me too (python 3.5.3 with Pycharm community, CuDA 8.0, windows 10, CuDNN 5.1 - works now)

**Comment by charles200000** [comment]
Worked for me too ! (python 3.5.3, windows 10, Cuda 8.0 and GTX 1080

**Comment by notilas** [comment]
CuDNN 5.1 (not 6.0) solved the problem!

**Comment by nectario** [comment]
I still get this error even though I have CUDNN 5.1 installed. What environment variables do I need specified?

**Comment by ChrisAntaki** [comment]
This issue seems to keep coming up for Windows users. Is there any way we could improve the error diagnosis/messaging, to give developers more guidance? For instance, if the cuDLL isn't found, the error message could suggest installing cuDNN 5.1, and adding it's bin directory to the path?

**Comment by ChrisAntaki** [comment]
@nectario Did you try adding the cuDNN's bin/ directory to your system's "Path" environmental variable?

![cuda-bin-path](https://user-images.githubusercontent.com/1216762/27255268-c3c4e9fc-534f-11e7-968e-af673f5e3493.png)

**Comment by nectario** [comment]
Let me do that! Than you!

**Comment by dE1l** [comment]
Check installed Microsoft Visual C++ 2015 Redistributable Update 3
You may install this DLL by downloading Microsoft Visual
C++ 2015 Redistributable Update 3 from this URL:
https://www.microsoft.com/en-us/download/details.aspx?id=53587

it's helped me

**Comment by Keson96** [comment]
@ChrisAntaki It worked for me, too! Thank you!

**Comment by chriswbarrett** [comment]
If you are using tensorflow 1.3 then you want to use the cuDNN v6.0 dll https://github.com/tensorflow/tensorflow/issues/7705

**Comment by Murali81** [comment]
@chriswbarrett  Thanks alot,I have wasted a day looking where I went wrong.Thank you!!!

**Comment by chenjw87** [comment]
windows server 2012 with 
tensorflow1.4 + CUDA 7.5 + cuDNN 6.0  fail to work
tensorflow1.4 + CUDA 7.5 + cuDNN 5.1  fail to work
how to address this problem??

much thanks!

**Comment by rohit-patel** [comment]
Had the same problem. Resolved using CUDA 9.0 and tf-nightly-gpu as suggested above. I have put up instructions here: https://github.com/rohit-patel/Install_Instructions-Win10-Deeplearning-Keras-Tensorflow
